learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,303 @@
+
+The quickest method to generate tenant tokens is [using one of our official SDKs](#generating-tenant-tokens-with-an-sdk). However, it is also possible to [generate the token without relying on any dependencies](#generating-tenant-tokens-without-a-meilisearch-sdk).
+
+## Generating tenant tokens with an SDK
+
+If you are developing an application that allows patients to search through medical records, it is crucial that each user can only see their own records and not anyone else's.
+
+The code in this example imports the SDK, creates a filter based on the current user's ID, and feeds that data into the SDK's `generateTenantToken` function. Once the token is generated, it is stored in the `token` variable:
+
+```js
+// placeholder code sample. Final version will be supplied by the integrations team
+// <CodeSamples id="tenant_token_guide_generate_sdk_1">
+const client = new Meilisearch({host: 'meilisearch'});
+
+const apiKey = '[an_api_key]';
+const currentUserID = '[placeholder_user_id]';
+
+const searchRules = {
+  'patient_medical_records': {
+    'filter': `user_id = ${currentUserID}`
+  }
+};
+
+const expiryDate = 1645461882;
+
+const token = client.generateTenantToken(searchRules, expiryDate, apiKey);
+```
+
+As you can see, there are three main elements necessary to generate a tenant token: [search rules](#search-rules), [expiry dates](#expiry-date), and [API keys](#api-keys). You can read more about each one these in [this guide's final section](#tenant-token-settings).
+
+### Using a tenant token with an SDK
+
+The resulting token, stored on the `token` variable, can then be sent back to the front-end and used to make queries that will only return results whose `user_id` attribute equals the current user's ID:
+
+```js
+// placeholder code sample. Final version will be supplied by the integrations team
+// <CodeSamples id="tenant_token_guide_sdk_search_1">
+const frontEndClient = new Meilisearch({
+  apiKey: token
+});
+
+const medicalRecordsIndex = frontEndClient.index('patient_medical_records');
+const search = medicalRecordsIndex.search('blood test');
+```
+
+Applications may use tenant tokens and API keys interchangeably when searching. For example, the same application might use a default search API key for queries on public indexes and a tenant token for logged-in users searching on private data.
+
+## Generating tenant tokens without a Meilisearch SDK
+
+Though [Meilisearch recommends using an official SDK](#generating-tenant-tokens-with-an-sdk), this is not a requirement. Since tenant tokens follow the JWT standard, you can use a number of [compatible third-party helper libraries](https://jwt.io/libraries). You may also skip all assistance and generate a token from scratch.
+
+The full process requires you to [create a token header](#creating-a-token-header), [prepare a data payload](#preparing-the-token-payload) with at least one set of search rules, and then [sign the token](#signing-a-token) with an API key.
```

Comment:
The structure I chose was dictated by two main ideas:

- show users some code as soon as possible;
- be as usage-agnostic and beginner-friendly as possible.

These two choices mean that the hardcore data part of the guide (the one containing exhaustive descriptions of the payload) gets pushed to the bottom of the page, under "Tenant token settings." 

Do you think more and clearer links to the last section of this document, `## Tenant token settings` could help?

Reply:
Yes, I think this could help! My point is more related to the idea of "explaining how to build a JWT is really necessary?". No problem if you think otherwise, just raising the hand about that! :D

Reply:
Ah, sorry, I misunderstood the first part!

In the @meilisearch/docs-team we tend to prioritise helping less experienced programmers, so there's always the possibility we become too didactic. One of my references for this guide was [Google's IoT token guide](https://cloud.google.com/iot/docs/how-tos/credentials/jwts#iot-core-jwt-nodejs), which does go over token creation, though.

I'll try to get a few more opinions before making a decision. Thanks for raising this point!